src/main/scala/com/comcast/xfinity/sirius/api/impl/SiriusFactory.scala

```diff
@@ -115,8 +117,7 @@ object SiriusFactory {
 
     // need to shut down the actor system and unregister the mbeans when sirius is done
     impl.onShutdown({
-      actorSystem.shutdown()
-      actorSystem.awaitTermination()
+      Await.ready(actorSystem.terminate(), Duration.Inf)
```

Comment:
I assume `awaitTermination` would also block forever?

Reply:
We are doing this in vinyldns...


```scala
        system.registerOnTermination(System.exit(0))

        // shut down ActorSystem
        system.terminate()
```

And it works fine.

Reply:
Correct, `awaitTermination()` also blocked forever.  I could change it to some arbitrary timeout if you'd prefer, or not block on termination.